model/common/src/icon4py/model/common/orchestration/decorator.py

```diff
@@ -0,0 +1,584 @@
+
+    def dace_specific_kwargs(
+        exchange_obj: decomposition.ExchangeRuntime,
+        offset_providers: dict[str, gtx.common.Connectivity],
+    ) -> dict[str, Any]:
+        return {
+            # connectivity tables
+            **{
+                connectivity_identifier(k): v.table
+                for k, v in offset_providers.items()
+                if hasattr(v, "table")
+            },
+            # GHEX C++ ptrs
+            "__context_ptr": expose_cpp_ptr(exchange_obj._context)
+            if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
+            else 0,
+            "__comm_ptr": expose_cpp_ptr(exchange_obj._comm)
+            if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
+            else 0,
+            **{
+                f"__pattern_{dim.value}Dim_ptr": expose_cpp_ptr(exchange_obj._patterns[dim])
+                if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
+                else 0
+                for dim in dims.global_dimensions.values()
+            },
+            **{
+                f"__domain_descriptor_{dim.value}Dim_ptr": expose_cpp_ptr(
+                    exchange_obj._domain_descriptors[dim].__wrapped__
+                )
+                if not isinstance(exchange_obj, decomposition.SingleNodeExchange)
+                else 0
+                for dim in dims.global_dimensions.values()
+            },
+        }
+
+    def modified_orig_annotations(
+        fuse_func: Callable, fuse_func_orig_annotations: dict[str, Any]
+    ) -> dict[str, Any]:
+        ii = 0
+        modified_fuse_func_orig_annotations = {}
+        for i, annotation_ in enumerate(fuse_func.__annotations__.values()):
+            if annotation_ is dace.compiletime:
+                modified_fuse_func_orig_annotations[
+                    list(fuse_func.__annotations__.keys())[i]
+                ] = None
+            else:
+                modified_fuse_func_orig_annotations[
+                    list(fuse_func.__annotations__.keys())[i]
+                ] = list(fuse_func_orig_annotations.values())[ii]
+                ii += 1
+        return modified_fuse_func_orig_annotations
+
+    def dace_symbols_concretization(
+        grid: icon_grid.IconGrid,
+        fuse_func: Callable,
+        fuse_func_orig_annotations: dict[str, Any],
+        args: Any,
+        kwargs: Any,
+    ) -> dict[str, Any]:
+        flattened_xargs_type_value = list(
+            zip(
+                list(fuse_func.__annotations__.values()),
+                list(args) + list(kwargs.values()),
+                strict=False,
+            )
+        )
+
+        def _concretize_symbols_for_dace_structure(dace_cls, orig_cls):
+            concretized_symbols = {}
+            for k_v in flattened_xargs_type_value:
+                if k_v[0] is not dace_cls:
+                    continue
+                for member in orig_cls.__dataclass_fields__.keys():
+                    for stride in [0, 1]:
```

Comment:
Can we generalize `for stride in [0, 1]` by looping on the number of dimensions?

Reply:
nice catch ;)